src/compressed_chunk.c

```diff
@@ -357,27 +344,37 @@ void Compressed_MRSerialize(Chunk_t *chunk, WriteSerializationCtx *sctx) {
                          (SaveStringBufferFunc)MR_SerializationCtxWriteBufferWrapper);
 }
 
-static char *ownedBufferFromMR(ReaderSerializationCtx *sctx, size_t *len) {
-    MRError *err;
-    size_t size = 0;
-    const char *temp = MR_SerializationCtxReadeBuffer(sctx, &size, &err);
-    char *ret = malloc(size);
-    memcpy(ret, temp, size);
-    if (len != NULL) {
-        *len = size;
-    }
-    return ret;
-}
-
 // this is just a temporary wrapper function that ignores error in order to preserve the common api
 static long long MR_SerializationCtxReadeLongLongWrapper(ReaderSerializationCtx *sctx) {
-    MRError *err; // TODO needs to call MR_ErrorCreate to allocate the error.
+    MRError *err;
     return MR_SerializationCtxReadeLongLong(sctx, &err);
 }
 
+static void Compressed_Deserialize_MR(Chunk_t **chunk,
+                                      void *ctx,
+                                      ReadUnsignedFunc readUnsigned,
+                                      ReadStringBufferFunc readStringBuffer) {
+    CompressedChunk *compchunk = (CompressedChunk *)malloc(sizeof(*compchunk));
+
+    compchunk->size = readUnsigned(ctx);
+    compchunk->count = readUnsigned(ctx);
+    compchunk->idx = readUnsigned(ctx);
+    compchunk->baseValue.u = readUnsigned(ctx);
+    compchunk->baseTimestamp = readUnsigned(ctx);
+    compchunk->prevTimestamp = readUnsigned(ctx);
+    compchunk->prevTimestampDelta = (int64_t)readUnsigned(ctx);
+    compchunk->prevValue.u = readUnsigned(ctx);
+    compchunk->prevLeading = readUnsigned(ctx);
+    compchunk->prevTrailing = readUnsigned(ctx);
+
+    size_t len;
+    compchunk->data = (uint64_t *)readStringBuffer(ctx, &len);
+    *chunk = (Chunk_t *)compchunk;
+}
+
 void Compressed_MRDeserialize(Chunk_t **chunk, ReaderSerializationCtx *sctx) {
-    Compressed_Deserialize(chunk,
-                           sctx,
-                           (ReadUnsignedFunc)MR_SerializationCtxReadeLongLongWrapper,
-                           (ReadStringBufferFunc)ownedBufferFromMR);
+    Compressed_Deserialize_MR(chunk,
```

Comment:
@OfirMos `Compressed_Deserialize_MR` used to be a a generic function for both MR/gears and RDB, but now we have 2 different function:
1. can we merge `Compressed_MRDeserialize` and `Compressed_Deserialize_MR` into one function?
2. maybe we should refactor the deserialize function into a macro to allow having 1 logic for both?